src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -431,7 +434,264 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
     /**
      * Sets attribute value.
      */
-    abstract protected function setAttributeValue(object $object, string $attribute, mixed $value, string $format = null, array $context = []);
+    abstract protected function setAttributeValue(object $object, string $attribute, $value, string $format = null, array $context = []);
+
+    protected function getFirstCollectionValueType(Type $type): ?Type
+    {
+        return $type->isCollection() ? $type->getCollectionValueTypes()[0] ?? null : null;
+    }
+
+    /**
+     * This should be done first when denormalizing attribute value.
+     *
+     * This function is not meant to be overriden, only used. If you might want to override validateAndDenormalize.
+     */
+    final protected function isNullDenormalization(Type $type, string $currentClass, string $attribute, mixed $data, ?string $format, array &$context): bool
```

Comment:
These functions always receive the same set of arguments in order to stay coherent and help developping. But you might want not to?